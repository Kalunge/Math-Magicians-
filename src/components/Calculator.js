/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';

export default class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const firstRow = ['AC', '+/-', '%'];
    const secondRow = [9, 8, 7];
    const thirdRow = [6, 5, 4];
    const fourthRow = [3, 2, 2];

    return (
      <div className="container">
        <div className="main mt-5 ">
          <div className="col-8 py-2 zero">
            <p className="px-2 pone">0</p>
          </div>
          <div className="row ">
            {firstRow.map((item) => (
              <div className="col-2 p-2 bg-light  " key={item}>
                <button type="button" className="p-2 btn col-sm ">
                  {item}
                </button>
              </div>
            ))}

            <div className="col-2 p-2 orange">
              <button type="button" className="p-2 btn col-sm ">
                รท
              </button>
            </div>
          </div>
          <div className="row ">
            {secondRow.map((item) => (
              <div className="col-2 p-2 bg-light  " key={item}>
                <button type="button" className="p-2 btn col-sm ">
                  {item}
                </button>
              </div>
            ))}
            <div className="col-2 p-2 orange">
              <button type="button" className="p-2 btn col-sm ">
                X
              </button>
            </div>
          </div>
          <div className="row ">
            {thirdRow.map((item) => (
              <div className="col-2 p-2 bg-light  " key={item}>
                <button type="button" className="p-2 btn col-sm ">
                  {item}
                </button>
              </div>
            ))}
            <div className="col-2 p-2 orange">
              <button type="button" className="p-2 btn col-sm ">
                -
              </button>
            </div>
          </div>
          <div className="row ">
            {fourthRow.map((item) => (
              <div className="col-2 p-2 bg-light  " key={item}>
                <button type="button" className="p-2 btn col-sm ">
                  {item}
                </button>
              </div>
            ))}
            <div className="col-2 p-2 orange">
              <button type="button" className="p-2 btn col-sm ">
                +
              </button>
            </div>
          </div>
          <div className="row ">
            <div className="col-4 p-2 bg-light div-border">
              <button type="button" className="p-2 btn col-sm ">
                0
              </button>
            </div>
            <div className="col-2 p-2 bg-light">
              <button type="button" className="p-2 btn col-sm ">
                .
              </button>
            </div>
            <div className="col-2 p-2 orange">
              <button type="button" className="p-2 btn col-sm ">
                =
              </button>
            </div>
          </div>
          <div className="col-4 " />
        </div>
      </div>
    );
  }
}
