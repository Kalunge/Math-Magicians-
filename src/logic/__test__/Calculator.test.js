import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Calculator from '../../components/Calculator';

describe('snapshot test for calculator Component', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot('Home');
  });
});

describe('simulate user event', () => {
  test('click', () => {
    render(<Calculator />);
    const button = screen.getByTestId('divide');
    userEvent.click(button);
    expect(onclick).toHaveBeenCalled;
  });
});
