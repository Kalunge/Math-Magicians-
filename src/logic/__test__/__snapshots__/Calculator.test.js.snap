// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for calculator Component renders correctly: Home 1`] = `
<div
  className="container row d-flex"
>
  <div
    className=" mt-5 math col-4"
  >
    <h1>
      Lets do some Math
    </h1>
  </div>
  <div
    className="main "
  >
    <div
      className="col-8 py-2 zero"
    >
      <span
        className="px-1 mr-5 pone"
      >
        0
      </span>
      <span
        className="px-1 mr-5 pone"
      />
      <span
        className="px-1 mr-5 pone"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="col-2 p-2 bg-light  "
      >
        <button
          className="p-2 btn col-sm "
          name="AC"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
      </div>
      <div
        className="col-2 p-2 bg-light  "
      >
        <button
          className="p-2 btn col-sm "
          name="+/-"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
      </div>
      <div
        className="col-2 p-2 bg-light  "
      >
        <button
          className="p-2 btn col-sm "
          name="%"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
      </div>
      <div
        className="col-2 p-2 orange"
      >
        <button
          className="p-2 btn col-sm "
          data-testid="divide"
          name="รท"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
    </div>
    <div
      className="row "
    >
      <div
        className="col-2 p-2 bg-light  "
      >
        <button
          className="p-2 btn col-sm "
          name={9}
          onClick={[Function]}
          type="button"
        >
          9
        </button>
      </div>
      <div
        className="col-2 p-2 bg-light  "
      >
        <button
          className="p-2 btn col-sm "
          name={8}
          onClick={[Function]}
          type="button"
        >
          8
        </button>
      </div>
      <div
        className="col-2 p-2 bg-light  "
      >
        <button
          className="p-2 btn col-sm "
          name={7}
          onClick={[Function]}
          type="button"
        >
          7
        </button>
      </div>
      <div
        className="col-2 p-2 orange"
      >
        <button
          className="p-2 btn col-sm "
          name="x"
          onClick={[Function]}
          type="button"
        >
          X
        </button>
      </div>
    </div>
    <div
      className="row "
    >
      <div
        className="col-2 p-2 bg-light  "
      >
        <button
          className="p-2 btn col-sm "
          name={6}
          onClick={[Function]}
          type="button"
        >
          6
        </button>
      </div>
      <div
        className="col-2 p-2 bg-light  "
      >
        <button
          className="p-2 btn col-sm "
          name={5}
          onClick={[Function]}
          type="button"
        >
          5
        </button>
      </div>
      <div
        className="col-2 p-2 bg-light  "
      >
        <button
          className="p-2 btn col-sm "
          name={4}
          onClick={[Function]}
          type="button"
        >
          4
        </button>
      </div>
      <div
        className="col-2 p-2 orange"
      >
        <button
          className="p-2 btn col-sm "
          name="-"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
    </div>
    <div
      className="row "
    >
      <div
        className="col-2 p-2 bg-light  "
      >
        <button
          className="p-2 btn col-sm "
          name={3}
          onClick={[Function]}
          type="button"
        >
          3
        </button>
      </div>
      <div
        className="col-2 p-2 bg-light  "
      >
        <button
          className="p-2 btn col-sm "
          name={2}
          onClick={[Function]}
          type="button"
        >
          2
        </button>
      </div>
      <div
        className="col-2 p-2 bg-light  "
      >
        <button
          className="p-2 btn col-sm "
          name={1}
          onClick={[Function]}
          type="button"
        >
          1
        </button>
      </div>
      <div
        className="col-2 p-2 orange"
      >
        <button
          className="p-2 btn col-sm "
          name="+"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
    </div>
    <div
      className="row "
    >
      <div
        className="col-4 p-2 bg-light div-border"
      >
        <button
          className="p-2 btn col-sm "
          name="0"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
      </div>
      <div
        className="col-2 p-2 bg-light"
      >
        <button
          className="p-2 btn col-sm "
          name="."
          onClick={[Function]}
          type="button"
        >
          .
        </button>
      </div>
      <div
        className="col-2 p-2 orange"
      >
        <button
          className="p-2 btn col-sm "
          name="="
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
    <div
      className="col-4 "
    />
  </div>
</div>
`;
